airbyte-integrations/connectors/source-github/unit_tests/integration/test_events.py

```diff
@@ -0,0 +1,158 @@
+        )
+
+    def teardown(self):
+        """Stops and resets HttpMocker instance."""
+        self.r_mock.__exit__()
+
+    def test_full_refresh_no_pagination(self):
+        """Ensure http integration, record extraction and transformation"""
+
+        self.r_mock.get(
+            HttpRequest(
+                url=f"https://api.github.com/repos/{_CONFIG.get('repositories')[0]}/events",
+                query_params={"per_page": 100},
+            ),
+            HttpResponse(json.dumps(find_template("events", __file__)), 200),
+        )
+
+        source = SourceGithub()
+        actual_messages = read(source, config=_CONFIG, catalog=_create_catalog())
+
+        assert len(actual_messages.records) == 2
+        assert all(("repository", "airbytehq/integration-test") in x.record.data.items() for x in actual_messages.records)
+
+    def test_full_refresh_with_pagination(self):
+        """Ensure pagination"""
+        self.r_mock.get(
+            HttpRequest(
+                url=f"https://api.github.com/repos/{_CONFIG.get('repositories')[0]}/events",
+                query_params={"per_page": 100},
+            ),
+            HttpResponse(
+                body=json.dumps(find_template("events", __file__)),
+                status_code=200,
+                headers={"Link": '<https://api.github.com/repos/{}/events?page=2>; rel="next"'.format(_CONFIG.get("repositories")[0])},
+            ),
+        )
+        self.r_mock.get(
+            HttpRequest(
+                url=f"https://api.github.com/repos/{_CONFIG.get('repositories')[0]}/events",
+                query_params={"per_page": 100, "page": 2},
+            ),
+            HttpResponse(
+                body=json.dumps(find_template("events", __file__)),
+                status_code=200,
+            ),
+        )
+        source = SourceGithub()
+        actual_messages = read(source, config=_CONFIG, catalog=_create_catalog())
+
+        assert len(actual_messages.records) == 4
+
+    def test_incremental_read(self):
+        """Ensure incremental sync.
+        Stream `Events` is semi-incremental, so all request  will be performed and only new records will be extracted"""
+
+        self.r_mock.get(
+            HttpRequest(
+                url=f"https://api.github.com/repos/{_CONFIG.get('repositories')[0]}/events",
+                query_params={"per_page": 100},
+            ),
+            HttpResponse(json.dumps(find_template("events", __file__)), 200),
+        )
+
+        source = SourceGithub()
+        actual_messages = read(
+            source,
+            config=_CONFIG,
+            catalog=_create_catalog(sync_mode=SyncMode.incremental),
+            state=StateBuilder()
+            .with_stream_state("events", {"airbytehq/integration-test": {"created_at": "2022-06-09T10:00:00Z"}})
+            .build(),
+        )
+        assert len(actual_messages.records) == 1
```

Comment:
It seems like there are other outcomes of incremental read that are probably interesting. Should we create other tests for `test_when_read_incrementally_then_emit_state_message` I don't see unit tests for Events so maybe we want to ensure the state value is correct too